forte/pipeline.py

```diff
@@ -798,6 +812,16 @@ def add(
             # This will ask the job to keep a copy of the gold standard.
             self.evaluator_indices.append(len(self.components))
 
+        if ref_name is None:
+            self._ref_names[ref_name] = []
+            self._ref_names[ref_name].append(len(self.components))
+        elif ref_name in self._ref_names.keys():
+            raise ValidationError(
+                f"This reference name {ref_name} already exists, please specify a new one"
```

Comment:
This branch is not tested, i.e. there is no unit test for the case where you add the same ref_name multiple times.